designs/0031-deprecation-schedule.md

```diff
@@ -0,0 +1,254 @@
+intentions, but also to let them tell us if our intentions need to be rethought.
+By communicating the changes and putting deadlines on them, we create a sort of
+community review period during which we can recieve feedback and course-correct
+if we have made a poor choice.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Deprecated minor features may be removed from Stan during minor version changes.
+Here, 'minor change' is restricted to be changes which can be done automatically
+through the use of the stanc3 canonicalizer, or a user-driven find and replace
+operation. By their very nature, minor changes are primarily syntactic. This is
+in contrast to 'major changes', which are by their nature semantic, and require
+human-in-the-loop reasoning to accommodate. Major changes, such the interface
+changes to the differential equation solvers, may be announced via deprecation
+warnings in minor versions, but will not be implemented/removed prior to a
+version 3.0.0.
+
+A minor feature must be deprecated for a minimum of 3 minor versions
+(approximately 1 year on the current release schedule) before removal. The exact
+timeline for a feature's removal will be decided during the development which
+would mark it as deprecated.
+
+A minor feature is deprecated and a candidate for future removal when:
+1. stanc emits a compile-time warning with the following 3 parts:
+    1. A description of the deprecated feature, such as "Use of <- for assignment
+       is deprecated".
+    2. An _action_ the user should take, such as "Use = instead". If stanc3 can
+       automate this for them, we should call that out, leading to a full
+       message like "Use = instead or run stanc3 --print-canonical to update
+       print an updated version of your code."
+    3. A timeline which includes a date or version after which this feature is
+       slated for removal, for example "This deprecation will expire in Stan
+       version 2.31.0 (expected Sept. 2022)".
+2. It has been added to the [documentation on
+   deprecation](https://mc-stan.org/docs/reference-manual/deprecated-features-appendix.html)
+   with the same 3 pieces of information, and additional information as needed
+   (such as any of: reasoning for removal, advantages of new behavior, ongoing
+   related changes, etc).
+3. Examples in the documentation have been updated to remove any examples of the
+   deprecated syntax, except for those referring to its deprecation. This change
+   may lag behind bullets 1 and 2, but must occur before the removal.
+
+Our goal is to only deprecate a feature when doing so provides a tangible
+benefit to users, such as improving error messaging or allowing new features to
+be added to the language while keeping a low maintenance burden. Changes should
+never take users (even those with many or large models) long to adjust to, and
+as few such changes should be made in each version as is reasonable.
```

Comment:
I don't see the point of spreading deprecations over multiple releases if they're ready.  We should only make deprecations that are reasonable, but given that, any number of them should be possible.  So I'd just drop line 90.